apps/judicial-system/backend/src/app/modules/defendant/defendant.service.ts

```diff
@@ -199,8 +201,18 @@ export class DefendantService {
   async updateByNationalId(
     caseId: string,
     defendantNationalId: string,
-    update: UpdateDefendantDto,
+    update: InternalUpdateDefendantDto,
   ): Promise<Defendant> {
+    // The reason we have a separate dto for this is because requests that end here
+    // are initiated by outside API's which should not be able to edit other fields
+    // Defendant updated originating from the judicial system should use the UpdateDefendantDto
+    // and go through the update method above using the defendantId.
+    // This is also why we set the isDefenderChoiceConfirmed to false here - the judge needs to confirm all changes.
```

Comment:
The reason I'm not comparing here to the current defender choice is because I know that the digital mailbox api that uses this function doesn't ever send us a request unless it's to change a defender choice. We don't get anything else from them. While the xrd-api that's using the subpoena update sends us a lot more stuff than that.